hir/src/module.rs

```diff
@@ -353,6 +353,17 @@ impl Module {
         self.globals.find(name).map(|gv| self.globals.get(gv))
     }
 
+    /// Find the first function in this module marked with the `entrypoint` attribute
+    pub fn entrypoint(&self) -> Option<FunctionIdent> {
```

Comment:
Should we forbid the module to add a function with an entrypoint attribute if the module already has one? To avoid multiple functions with an entrypoint attribute. I see the downside of needing an extra step to swap an entrypoint function, but other than that, it seems fine. Although it might be hard to enforce this on transformation passes.

Reply:
I think we can make this a job for the linker, since the same problem exists if there are entrypoints declared in multiple modules. Additionally, `Program` allows one to manually override what function is the entrypoint, and in that situation, we don't need to raise an error if there are multiple functions with the `entrypoint` attribute, since we're being told which function to use. So if we have the linker handle this, we can keep the check in one place, and also be able to handle the case where an entrypoint was specified manually.

We could add a module validation rule though that raises a warning if there are multiple entrypoints in a module, since even if it doesn't end up being an error due to specifying an entrypoint, it doesn't make sense to define multiple entrypoints in a single module anyway (unless we add something like Rust's `cfg` attributes, and you have multiple entrypoints that are mutually exclusive based on configuration).